packages/transformers/html/src/dependencies.js

```diff
@@ -123,6 +124,20 @@ export default function collectDependencies(
 
     seen.add(node);
 
+    if (tag === 'img' && attrs.src && (attrs.width || attrs.height)) {
+      const src = url.parse(attrs.src, true);
+      if (!src.protocol && !src.query.width && !src.query.height) {
+        if (attrs.width) {
+          src.query.width = attrs.width;
+        }
+        if (attrs.height) {
+          src.query.height = attrs.height;
+        }
+
+        attrs.src = url.format({pathname: src.pathname, query: src.query});
```

Comment:
Was surprised to find that `url.format(src)` doesn't use the modified version. I know that the Node URLs are deprecated in favour of the WHATWG form, but that doesn't support relative URLs (which is the only case we want to process here).

So... this works, but there's probably a better way of doing this.